server/container_create_linux.go

```diff
@@ -867,6 +867,8 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrfactory.Cont
 	ociContainer.SetSpec(specgen.Config)
 	ociContainer.SetMountPoint(mountPoint)
 	ociContainer.SetSeccompProfilePath(seccompRef)
+	// Set the architecture of the image for container
+	ociContainer.SetArch(containerInfo.Config.Platform.Architecture)
```

Comment:
TODO: do we need to account for os as well? https://github.com/containers/podman/pull/16434/files#diff-c15f51066222fb402fefe74faa0f789de498024988e4a0414f5713f3b40bdf33R628

Reply:
I don't think so. We are already extracting the Arch from the [Platform](https://github.com/opencontainers/image-spec/blob/main/specs-go/v1/config.go#L101) struct.